credentials/credentials.go

```diff
@@ -334,3 +336,31 @@ func cloneTLSConfig(cfg *tls.Config) *tls.Config {
 
 	return cfg.Clone()
 }
+
+// RequestInfo contains request data attached to the context passed to GetRequestMetadata calls.
+//
+// This API is experimental.
+type RequestInfo struct {
+	// The method passed to Invoke or NewStream for this RPC. (For proto methods, this has the format "/some.Service/Method")
+	Method string
+}
+
+// requestInfoKey is a struct to be used as the key when attaching a RequestInfo to a context object.
+type requestInfoKey struct{}
+
+// RequestInfoFromContext extracts the RequestInfo from the context.
+//
+// This API is experimental.
+func RequestInfoFromContext(ctx context.Context) RequestInfo {
```

Comment:
Bah, I missed this before, sorry: it's typical to also return an `ok bool` here, in case the `RequestInfo` is not present in `ctx`.  Please add this.  The comment can be updated to match `peer.FromContext`'s wording by simply adding "if it exists".

Reply:
Done.